### PR TITLE
[FIX] web_editor: properly fetching of colorpicker template

### PR DESCRIPTION
**Pull request by Goaman**
As described in #129974, the colorpicker cannot cache a promise protected by `useService`. #129974 fixes the issue by through non idiomatic practice (calling a service that is not protected by `useService`). This commit fixes the issue idiomatically.

This commit caches the promise directly from a service and removes the caching code from the `ColorPalette`.

task-3446691

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js [hunk 1]
```diff
@@ -97,7 +71,7 @@ export class ColorPalette extends Component {
         this.init();
         onWillStart(async () => {
             if (this.props.getTemplate) {
-                await ColorPalette.loadDependencies(this.props.getTemplate);
+                this.colorpickerArch = await this.props.getTemplate();
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js:74]: Ouch, I thought I had you rename all those "arch" things to "template"... looks like we missed some of them... :/

**Review by dmo-odoo — COMMENTED** [review]
@aab-odoo Is this what you had in mind ? It seems a tad overkill to me to create a new service just for the sake of handling the colorpicker template, but if you meant something else then I didn't understand what you mean... ^^'

**Comment by aab-odoo** [comment]
> @aab-odoo Is this what you had in mind ? It seems a tad overkill to me to create a new service just for the sake of handling the colorpicker template, but if you meant something else then I didn't understand what you mean... ^^'

The problem is that when you do an rpc in a component, it is tied to that component instance (via useService). In this case, the rpc is global (you want to do it only once), not specific to a component instance. The natural of doing so in wowl is to define a service, which is never destroyed (it is started at webclient startup and never destroyed). So the service can keep the rpc promise in its closure (it can wait for the first request to actually do the rpc), ensure that it is done only once, and the promise won't be pending forever as it won't be tied to any component instance. Note that I tried the service solution, and strictly speaking I think it was ok. The problem was with the tests, as a lot of them would require the service to be available (but in tests, we kind of whitelist services we need, so that one wasn't there).

**Comment by dmo-odoo** [comment]
@robodoo r+

**Comment by dmo-odoo** [comment]
@robodoo retry

**Pull request closed without merging**